src/Symfony/Component/Console/Helper/ProgressIndicator.php

```diff
@@ -0,0 +1,292 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Helper;
+
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ */
+class ProgressIndicator
+{
+    private $output;
+    private $startTime;
+    private $format;
+    private $message = null;
+    private $indicatorValues = array('-', '\\', '|', '/');
+    private $indicatorCurrent = 0;
+    private $indicatorChangeInterval = 100;
+    private $indicatorUpdateTime;
+    private $lastMessagesLength = 0;
+
+    private static $formatters;
+    private static $formats;
+
+    /**
+     * @param OutputInterface $output
+     */
+    public function __construct(OutputInterface $output)
+    {
+        $this->output = $output;
+        $this->startTime = time();
+
+        $this->setFormat($this->determineBestFormat());
+    }
+
+    /**
+     * @param string $format
+     */
+    public function setFormat($format)
+    {
+        $this->format = self::getFormatDefinition($format);
+    }
+
+    /**
+     * @param int $milliseconds
+     */
+    public function setIndicatorChangeInterval($milliseconds)
+    {
+        $this->indicatorChangeInterval = $milliseconds;
+    }
+
+    /**
+     * @param string|null $message
+     */
+    public function setMessage($message)
+    {
+        $this->message = $message;
+
+        $this->display();
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getMessage()
```

Comment:
Will this cause issues in 5.3 (https://github.com/symfony/symfony/pull/12119/files#diff-c69956b5e0c001490b2ebfa0b9e2db1aR268) or will this only be in 3.0?